presto-main/src/test/java/com/facebook/presto/operator/TestHashAggregationOperator.java

```diff
@@ -186,8 +185,8 @@ public void testHashAggregation(boolean hashEnabled, long memoryLimitBeforeSpill
         assertOperatorEqualsIgnoreOrder(operatorFactory, driverContext, input, expected, hashEnabled, Optional.of(hashChannels.size()));
     }
 
-    @Test(dataProvider = "hashEnabled")
-    public void testHashAggregationWithGlobals(boolean hashEnabled)
+    @Test(dataProvider = "hashEnabledAndMemoryLimitBeforeSpillValues")
+    public void testHashAggregationWithGlobals(boolean hashEnabled, long memoryLimitBeforeSpill, long memoryLimitForMergeWithMemory)
```

Comment:
you're right. we don't test that and now it's broken. I added temporary TODO note to unblock review process, to be done _before_ PR gets merged.